microcache.go

```diff
@@ -134,7 +134,7 @@ type Config struct {
 }
 
 // New creates and returns a configured microcache instance
-func New(o Config) Microcache {
+func New(o Config) *microcache {
```

Comment:
It can be annoying to use an unexported return value (for example I wouldn't be able to store it as a member on a struct)

Reply:
It can still be referenced by the interface, `microcache.Microcache`